src/checker.rs

```diff
@@ -136,13 +137,21 @@ impl<'a> Checker<'a> {
         timeout: Option<Duration>,
         verbose: bool,
         progress_bar: Option<&'a ProgressBar>,
+        basic_auth: Option<String>,
     ) -> Result<Self> {
-        let mut headers = header::HeaderMap::new();
+        let mut headers = HeaderMap::new();
         // Faking the user agent is necessary for some websites, unfortunately.
         // Otherwise we get a 403 from the firewall (e.g. Sucuri/Cloudproxy on ldra.com).
         headers.insert(header::USER_AGENT, HeaderValue::from_str(&user_agent)?);
         headers.insert(header::TRANSFER_ENCODING, HeaderValue::from_str("chunked")?);
 
+        // Add basic header if exist
+        if let Some(b) = basic_auth {
+            let params: Vec<&str> = b.split(':').collect();
+            let auth = Authorization::basic(params[0], params[1]);
```

Comment:
This could fail in case there is just one element in `params`, e.g. if `b` is `hello` without a colon. Consider the snippet [here](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=cc9ad42d4a093a9c4139302431a61af3).

I'd prefer to do the parsing outside of the checker and pass in the `Authorization` header as part of `custom_headers`. 

The parsing function could be in `main.rs` for now and have a signature like `fn parse_auth(auth: &str) -> Authorization`. This way we could add some isolated unit tests just for the parsing.

What do you think?

Reply:
Yehh sound great to me and can be useful for another types of auths in the future